v-next/hardhat/src/internal/cli/init/project-creation.ts

```diff
@@ -0,0 +1,238 @@
+import type { PackageJson } from "@nomicfoundation/hardhat-utils/package";
+
+import path from "node:path";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import { ensureError } from "@nomicfoundation/hardhat-utils/error";
+import {
+  exists,
+  readJsonFile,
+  writeJsonFile,
+  writeUtf8File,
+} from "@nomicfoundation/hardhat-utils/fs";
+import chalk from "chalk";
+
+import { getHardhatVersion } from "../../utils/package.js";
+
+const HARDHAT_NAME = "Hardhat";
+const HARDHAT_PACKAGE_NAME = "hardhat";
+
+// TODO: test that is testing the last version of solidity
+export const EMPTY_HARDHAT_CONFIG = `/** import type { HardhatUserConfig } from "@nomicfoundation/hardhat-core/config"; */
+
+export default {
+  solidity: "0.8.24",
+};
```

Comment:
Something like [this](https://github.com/NomicFoundation/hardhat/pull/5425/commits/6f99e6d08438f4ef58555b862ce2c30cc147b8a0)?

Reply:
Hmm, no, that's not what I meant. You moved the logic to a different file, but it's still a variable with the Hardhat config code within a string. I meant having an actual file with the code, so we can easily update it, ensure it's well formatted, and maintain it more efficiently. Then you can read that file here and copy it.

Reply:
I see what you mean. The only downside with the suggested approach is that the solidity version will have to be updated manually. While with the current approach the version update will happen automatically. 
Since it's a very short example file I'd keep this approach, wdyt?

Reply:
That's a good point. However, we'll need to manually update the Solidity version for the JS/TS sample projects anyway, right? This makes me think that we should export a constant with the current version from Hardhat and use it there. But that's something to discuss. I'm good if you want to leave it the way it is for now.